dotnet/src/SemanticKernel/Planning/PlannerConfigBase.cs

```diff
@@ -21,7 +30,49 @@ public abstract class PlannerConfigBase
     public HashSet<string> ExcludedFunctions { get; } = new();
 
     /// <summary>
-    /// Delegate to get the prompt template string.
+    /// When using <see cref="Memory"/> to get relevant functions,
+    /// this list of functions will be included regardless of relevancy.
     /// </summary>
-    public Func<string>? GetPromptTemplate { get; set; } = null;
+    public HashSet<(string, string)> IncludedFunctions { get; } = new();
+
+    /// <summary>
+    /// Semantic memory to use for function lookup (optional).
+    /// This property will be ignored if <see cref="GetAvailableFunctionsAsync"/> is set.
+    /// </summary>
+    public ISemanticTextMemory Memory { get; set; } = NullMemory.Instance;
+
+    /// <summary>
+    /// The maximum number of relevant functions to include in the plan.
+    /// </summary>
+    /// <remarks>
+    /// Limits the number of relevant functions as result of semantic
+    /// search included in the plan creation request.
+    /// <see cref="IncludedFunctions"/> will be included
+    /// in the plan regardless of this limit.
+    /// </remarks>
+    public int MaxRelevantFunctions { get; set; } = 100;
+
+    /// <summary>
+    /// The minimum relevancy score for a function to be considered
+    /// </summary>
+    /// <remarks>
+    /// Depending on the embeddings engine used, the user ask, the step goal
+    /// and the functions available, this value may need to be adjusted.
+    /// For default, this is set to null to exhibit previous behavior.
+    /// </remarks>
+    public double? RelevancyThreshold { get; set; }
```

Comment:
What do you think about grouping config properties related to memory in a MemoryConfig object? It's not immediately obvious to me that this is related to memory.

Also, for those not familiar with this, it would be good to describe what a relevancy score is and how it ranges from 0 to 1.